engine/src/integration_tests.rs

```diff
@@ -5561,6 +5561,25 @@ fn test_manual_bridge() {
     do_run_test_manual("", hdr, rs, &[], None).unwrap();
 }
 
+#[test]
+fn test_issue486() {
+    let hdr = indoc! {"
+        namespace a {
+            namespace spanner {
+                class Key;
+            }
+        } // namespace a
+        namespace spanner {
+            class Key {
+                public:
+                    bool b(a::spanner::Key &);
```

Comment:
It's possible that the test will pass if you change this line to `inline bool b(a::spanner::Key &) {}`

Reply:
Actually no it won't, the problems seem to be genuine problems confusing `Key` and `spanner::Key`